src/main/java/mate/jdbc/impl/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,114 @@
+                        connection.prepareStatement(query)) {
+            getManufacturerStatement.setLong(1, id);
+            ResultSet resultSet = getManufacturerStatement.executeQuery();
+            if (resultSet.next()) {
+                optionalManufacturer = getManufacturer(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get manufacturer from "
+                    + "database by id: " + id + ".", e);
+        }
+        return optionalManufacturer;
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        String query = "SELECT * FROM manufacturers WHERE is_deleted = false;";
+        List<Manufacturer> manufacturers = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllManufacturersStatement =
+                        connection.prepareStatement(query)) {
+            ResultSet resultSet = getAllManufacturersStatement.executeQuery();
+            while (resultSet.next()) {
+                Optional<Manufacturer> optionalManufacturer = getManufacturer(resultSet);
+                optionalManufacturer.ifPresent(manufacturers::add);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all manufacturers from database.", e);
+        }
+        return manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String query = "UPDATE manufacturers SET name = ?, country = ? WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateManufacturerStatement =
+                        connection.prepareStatement(query)) {
+            updateManufacturerStatement.setString(1, manufacturer.getName());
+            updateManufacturerStatement.setString(2, manufacturer.getCountry());
+            updateManufacturerStatement.setLong(3, manufacturer.getId());
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update manufacturer: "
+                    + manufacturer + " in database.", e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE manufacturers SET is_deleted = true WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteManufacturerStatement =
+                        connection.prepareStatement(query)) {
+            deleteManufacturerStatement.setLong(1, id);
+            return deleteManufacturerStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete manufacturer "
+                    + "from database by id:" + id + ".", e);
+        }
+    }
+
+    private Optional<Manufacturer> getManufacturer(ResultSet resultSet) throws SQLException {
```

Comment:
```suggestion
    private Manufacturer getManufacturer(ResultSet resultSet) throws SQLException {
```
just return manufacturer, you  don't need to use Optional here

Reply:
Agree, this method never return null